app/src/main/java/com/aletheiaware/perspective/android/ui/WorldSelectActivity.java

```diff
@@ -131,7 +150,7 @@ public void onSelect(World world) {
             PuzzleAdapter puzzleAdapter = new PuzzleAdapter(this, world, puzzleStars.get(name), new PuzzleAdapter.Callback() {
                 @Override
                 public void onSelect(String world, int puzzle) {
-                    if (dialog != null && dialog.isShowing()) {
+                    if (dialog.isShowing()) {
```

Comment:
FTF: I'd like to suggest that you strive to use @NonNull/@Nullable as much as possible and minimize methods that return null. For example, see comment on getNextWorld.

Reply:
Done